src/storage/store/rel_table_data.cpp

```diff
@@ -499,18 +537,18 @@ void RelTableData::prepareCommitCSRNGWithSliding(Transaction* transaction,
     }
 }
 
-std::unique_ptr<ColumnChunk> RelTableData::slideCSROffsetChunk(
-    ColumnChunk* csrOffsetChunk, CSRRelNGInfo* relNodeGroupInfo) {
-    auto slidedCSRChunk =
+std::pair<std::unique_ptr<ColumnChunk>, std::unique_ptr<ColumnChunk>>
+CSRRelTableData::slideCSRAuxChunks(
+    ColumnChunk* csrOffsetChunk, ColumnChunk* csrLengthChunk, CSRRelNGInfo* relNodeGroupInfo) {
+    auto slidedCSROffsetChunk =
+        ColumnChunkFactory::createColumnChunk(LogicalType::INT64(), enableCompression);
+    auto slidedCSRLengthChunk =
         ColumnChunkFactory::createColumnChunk(LogicalType::INT64(), enableCompression);
     int64_t currentCSROffset = 0;
     auto currentNumSrcNodesInNG = csrOffsetChunk->getNumValues();
     auto newNumSrcNodesInNG = currentNumSrcNodesInNG;
     for (auto offsetInNG = 0u; offsetInNG < currentNumSrcNodesInNG; offsetInNG++) {
-        int64_t numRowsInCSR = offsetInNG == 0 ?
-                                   csrOffsetChunk->getValue<offset_t>(offsetInNG) :
-                                   csrOffsetChunk->getValue<offset_t>(offsetInNG) -
-                                       csrOffsetChunk->getValue<offset_t>(offsetInNG - 1);
+        int64_t numRowsInCSR = csrLengthChunk->getValue<offset_t>(offsetInNG);
```

Comment:
use auto?

Reply:
I actually want to explicitly set these variables to `int64_t` as I can asserting them to be larger or equal to 0.